### DISCUSSION TITLE
ADS1115 Error

### DISCUSSION THREAD

**Discussion by rcowan75** [nebhead/PiFire#103] [Q&A]
![Screenshot_20220527-121720](https://user-images.githubusercontent.com/100400562/170781970-32a0f1ca-e462-4d45-871a-d5d5b5ab6a22.png)
I'm trying to get my PiFire project started up. I keep getting this error related to ADS1115. I have tried a couple boards (incase the first was bad) but I get the same message. I used an ohm meter to check the connections between the pins on the ADS and Pi. They look good. I DON'T have probes attached yet but I don't think that it what this error is telling me, correct? Any ideas on what to do next?

**Comment by nebhead** [comment]
I'm about to step out the door for the long weekend, but just saw this.  It's more likely something went bad during the installation process and perhaps the ADS1115 module wasn't installed properly?  Maybe try to download the logs (see the admin page) and post the control.err.log and control.out.log here.

**Reply by nebhead** [reply]
Also... It might be worth trying to enable I2C from sudo raspi-config in case that is the issue.

**Reply by nebhead** [reply]
@rcowan75 

Checking back in to see if you are still experiencing an issue.

**Comment by rcowan75** [comment]
Thanks for checking in. I should have replied but still don't have a full answer so I hadn't yet. Enabling I2C fixed the ads1115 error. However, I'm still not cooking. The relays that used to work is no longer working. When I go to manual mode, the LEDs are lighting up, indicating that it is trying to switch, however the relay is not actually closing (confirmed with meter). I still need to double check the high/low switch setting to see if that is the issue.

**Reply by nebhead** [reply]
> Thanks for checking in. I should have replied but still don't have a full answer so I hadn't yet. Enabling I2C fixed the ads1115 error. 

No problem!  Glad this worked for you.  I'm going to have to look into why it's not getting enabled during installation.  Should be part of the install script to edit the config.txt to enable I2C.  Anyway, glad to hear you got it going.  

> However, I'm still not cooking. The relays that used to work is no longer working. When I go to manual mode, the LEDs are lighting up, indicating that it is trying to switch, however the relay is not actually closing (confirmed with meter). I still need to double check the high/low switch setting to see if that is the issue.

What relay board are you using?  Is it possible you have the 12V version?  That might turn on the LED with 5V, but might not be enough to move the coil on the relay (if it's a mechanical relay).

**Comment by rcowan75** [comment]
Good point. I know I ordered the 5V version, not sure what I got as I don't see any marking of voltage. I ordered the linked (in your documentation) relay board (https://www.amazon.com/gp/product/B07BLL4HFD/ref=ppx_yo_dt_b_asin_title_o01_s01?ie=UTF8&th=1). I just double checked my order and I did order the 5V version.

It has worked in the past. Before I the I2C was enabled, it would go into a demo mode and that mode turned on/off fans, auger, etc.

**Reply by nebhead** [reply]
OK, great.  Then you should have your relays set to high level trigger for that one.  Make sure when you run the wizard to set the drop down for the trigger level to high.

![image](https://user-images.githubusercontent.com/19876886/171666278-f08783c3-ee5f-49b2-9345-2cd91c708d69.png)

**Comment by rcowan75** [comment]
I have ran through the wizard a few times, selecting Active High every time but still no luck. The relays simply aren't closing.

One thing I found odd is that every time I went through the wizard, the trigger level was set to Active Low and I had to switch to Active High. Makes me wonder if it's not setting it during the config process? Should the wizard show the current active selection? That seems to be the case with the display.

**Reply by nebhead** [reply]
> I have ran through the wizard a few times, selecting Active High every time but still no luck. The relays simply aren't closing.

Very strange, but when you were using the prototype probe sensor (instead of the ADS115), you are seeing the outputs of the relay actually switch properly?  Is it possible to confirm this using a multi-meter?  Also, which PCB are you using?  Are you certain that the relay board is getting 5V?  

> One thing I found odd is that every time I went through the wizard, the trigger level was set to Active Low and I had to switch to Active High. Makes me wonder if it's not setting it during the config process? Should the wizard show the current active selection? That seems to be the case with the display.

Oh, the wizard just always displays defaults, not what is currently selected.  That would actually be a good idea, but the way it's currently designed, it will always bring up defaults.  So what you are seeing is expected behavior.

**Reply by nebhead** [reply]
Also - Double check your wiring here: https://nebhead.github.io/PiFire/hardware#ac-components--relay-schematic 

Since you are using the solid-state relay, ensure you are using the stand-alone wiring diagram.

**Reply by rcowan75** [reply]
> Very strange, but when you were using the prototype probe sensor (instead of the ADS115), you are seeing the outputs of the relay actually switch properly? Is it possible to confirm this using a multi-meter? Also, which PCB are you using? Are you certain that the relay board is getting 5V?

I switched back to using the prototype probe sensor to see if I could get any of the relays to close. No luck. Nothing works in that mode either (but did in the past). The LEDs in the replay board that indicate on/off of the relays DO work so I think the wiring between the pifire model and the relay board is correct. I verified with a meter that the relays are not closing. I think I may have passed too much current through the relay board and fried it. That's my current guess anyway. I noticed that the relay panel is rated at 2A. That seems low, especially when the ignitor is 300w alone. Has that relay board worked well for others? (https://www.amazon.com/gp/product/B07BLL4HFD/ref=ppx_yo_dt_b_asin_title_o01_s01?ie=UTF8&th=1)

My fried board theory comes from the fact that I failed to install a fuse protecting the PiFire AND that I was tripping the GFI sometimes when it was in prototype probe sensor mode. I bought this grill as a non-working project. I bet I need a new ignitor and new relay board.

**Reply by nebhead** [reply]
> I switched back to using the prototype probe sensor to see if I could get any of the relays to close. No luck. Nothing works in that mode either (but did in the past). The LEDs in the replay board that indicate on/off of the relays DO work so I think the wiring between the pifire model and the relay board is correct. I verified with a meter that the relays are not closing. I think I may have passed too much current through the relay board and fried it. That's my current guess anyway. I noticed that the relay panel is rated at 2A. That seems low, especially when the ignitor is 300w alone. Has that relay board worked well for others? (https://www.amazon.com/gp/product/B07BLL4HFD/ref=ppx_yo_dt_b_asin_title_o01_s01?ie=UTF8&th=1) My fried board theory comes from the fact that I failed to install a fuse protecting the PiFire AND that I was tripping the GFI sometimes when it was in prototype probe sensor mode. I bought this grill as a non-working project. I bet I need a new ignitor and new relay board.

Before you toss out the board, I realized that when testing continuity on the relay with a meter, you'll want to make sure there is actually a load attached.  With solid state relays, they do require some minimum current for them to work.  Testing DC voltages across the outputs aren't going to work.  (source: https://forum.digikey.com/t/can-i-test-solid-state-relays-ssrs-with-a-multi-meter/546).  So, while it's connected to AC power, measure AC voltage in both the on and off state.  Please be very careful not to electrocute yourself!  Don't accidentally short anything.  Perform testing at your own risk.  

Also, if you are tripping the GFI, there is a pretty good chance your igniter might be toast.  You can usually test this with your Ohm setting on your multimeter.  If you are getting a very high resistance (>1000 Ohms) or a short, then your igniter is probably bad.  You can test your other components in a similar way, to see if any have a short and are bad.  (https://youtu.be/27vnkZ7hpZs?t=50)  A short will definitely trip the GFI.

**Reply by nebhead** [reply]
I should add, that my SSR is also rated for 2A at 240V (consider that would probably be around 4A at 120V) - and it looks nearly identical to the Amazon one you posted.  I have had no issues with mine.  The igniter is the component that draws the most current and when mine is active, it's drawing around 4A from the wall for all of the components including the fan, auger and Raspberry Pi Zero W.

**Reply by rcowan75** [reply]
I tested the relay similar to how Digikey link showed earlier and it was not passing voltage. A new relay board and ignitor are on the way. I really hope that takes care of it. Oh! I will add a fuse also so hopefully I won't have to buy another board, maybe just a few more fuses to figure this out! :-)

**Reply by nebhead** [reply]
Oh, well bummer I guess it must have gotten damaged.  Perhaps if the igniter was shorting, then it overloaded the ssr.  I've got my fingers crossed for the new one.  If you have any mechanical relay boards hanging around, you can try those temporarily to make sure that works.  If not, just wait and see.  Good luck!

**Reply by rcowan75** [reply]
New relay board and ignitor showed up. I first installed a 5A fuse and new relay board. Relays worked again and I could turn on everything HOWEVER, the ignitor would trip the GFI. I installed new ignitor and no more GFI tripping!! I was able to fire up and run the grill. I have some work to do on the PID side, they temp isn't consistent but it works!! Thanks for the help!!
On the PID side, I think most people are using PiFire with a Traeger, I'm using it with a Sawtooth. I'm not sure what the differences are but there must be some since it seems the PID settings seem to work well out of the box for most. Did the PID code come from PiSmoker or Heatermeter? I think I read it came from one of them but can't find it now. I have a Heatermeter also so I'm fairly familiar with tuning that.... please say it came from there. :-)

**Reply by nebhead** [reply]
> New relay board and ignitor showed up. I first installed a 5A fuse and new relay board. Relays worked again and I could turn on everything HOWEVER, the ignitor would trip the GFI. I installed new ignitor and no more GFI tripping!! I was able to fire up and run the grill. I have some work to do on the PID side, they temp isn't consistent but it works!! Thanks for the help!! 

Happy to help and glad to hear it's working!  I knew we would get there. :) 

> On the PID side, I think most people are using PiFire with a Traeger, I'm using it with a Sawtooth. I'm not sure what the differences are but there must be some since it seems the PID settings seem to work well out of the box for most. Did the PID code come from PiSmoker or Heatermeter? I think I read it came from one of them but can't find it now. I have a Heatermeter also so I'm fairly familiar with tuning that.... please say it came from there. :-)

Actually, the PID is from the PiSmoker project by @dborello.  I lifted all of the PID code (nearly exactly) from that project and @weberbox has made some tweaks to it.  But feel free to take a look and do some further adjustments.  Some of the PID swings can be due to your smoker characteristics (size/volume, thermal transfer, etc.), the ambient temperature and the type of pellets.  But given enough time, I find the PID can hold a temperature quite well.  Sometimes it does take up to 30 minutes or so for things to home in, sometimes much faster, but just depends on all the variables.   We have many people using PiFire on Pit Boss grills as well (including @weberbox I think), which seems to work well.  But as always, YMMV.

**Comment by rcowan75** [comment]
This makes me very happy. I have burned through 3 bags getting the PID settings correct but this is the first cook and it's doing very well. Thanks for the help! ![Sawtooth](https://user-images.githubusercontent.com/100400562/176246591-6ace2a9b-218f-41e7-a5f2-b1a53e7139fa.JPG)

**Reply by nebhead** [reply]
Awesome!!  Care to share your settings and your configuration that you settled on?

**Reply by rcowan75** [reply]
Sure. The only settings I modified were the PID Settings. They are:
![Sawtooth2](https://user-images.githubusercontent.com/100400562/176320815-28db9c49-3be3-49b3-839f-9d8a3b0d7362.JPG)
I'm using these settings on a Sawtooth SPG-610. (https://www.sawtoothpelletgrills.com/product/spg-610-sawtooth-pellet-grill-fully-loaded/https://www.sawtoothpelletgrills.com/product/spg-610-sawtooth-pellet-grill-fully-loaded/)

I found that the information I used when tuning my Heater Meter was helpful in tuning the PID on the PiFire also. That information is located here: https://github.com/CapnBry/HeaterMeter/wiki/PID-Tuning

**Reply by nebhead** [reply]
Thank you for sharing!
